### PR TITLE
UI Review: Snapshot view

### PR DESCRIPTION
**Pull request by utkarshcmu**

### CODE REVIEW

**Comment by utkarshcmu** [comment]
@bergquist / @torkelo This is Work in Progress with most of the work done. Please review the UI / functionality of this PR. Should we add anything else on the UI for Snapshots?

Things left to do to get this PR merged:
1. Correct angular variables names according to the convention.
2. Write tests. :)
I will try to finish the above 2 ASAP.

**Comment by torkelo** [comment]
looks like a good start :)

Some minor things that need to change:
1) Put routes in routes file, and use isolated scope and controllerAs, https://github.com/grafana/grafana/blob/master/public/app/core/routes/all.js#L141

2) Write controller like this, https://github.com/grafana/grafana/blob/master/public/app/features/apps/edit_ctrl.ts

**Comment by mattttt** [comment]
This is looking great, a couple remarks:

- This will be a sub-nav link under Dashboards, which is a new feature that's currently in master. I havent added any links to it myself yet, but Im sure @torkelo can offer some color.
- Subnav links do not use icons, but for general knowledge, we use gf-icon-snap-multi. The camera (fa-camera-retro ) is not the correct icon.

Love this, thanks @utkarshcmu.

**Comment by utkarshcmu** [comment]
Only parts left now are:
1. Writing tests
2. Implementing Subnav functionality as described by @mattttt

File: public/app/features/snapshot/snapshot_ctrl.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+///<reference path="../../headers/common.d.ts" />
+
+import angular from 'angular';
+import _ from 'lodash';
+
+export class SnapshotsCtrl {
+
+  /** @ngInject */
+  constructor(backendSrv, $scope) {
+    $scope.init = function() {
```
Comment (torkelo) [inline public/app/features/snapshot/snapshot_ctrl.ts]: don't use scope, look again at https://github.com/grafana/grafana/blob/master/public/app/features/apps/edit_ctrl.ts 

You will need to update the html partial to bind the controller

Comment (utkarshcmu) [inline public/app/features/snapshot/snapshot_ctrl.ts]: I tried the way you are telling but then I got stuck at this.AppEvent. No worries, will try once more.

Comment (torkelo) [inline public/app/features/snapshot/snapshot_ctrl.ts]: yea sorry, forgot to mention that, you will have to use $rootScope to access that function, like this:
 $rootScope.onAppEvent('zoom-out', () => this.zoom(2), $scope);
    
The reason for wanting to write code like this is to ease migration to angular2  (angular2 has completely removed the concept of scopes), code like this is also less coupled trough hidden scope prototype chains

Comment (utkarshcmu) [inline public/app/features/snapshot/snapshot_ctrl.ts]: Transformed the code for the ease of migration to angularjs2. :)

**Comment by utkarshcmu** [comment]
@torkelo This PR is ready from my side as I am not sure about how @mattttt wants to place snapshots as subnav to dashboards.

**Comment by utkarshcmu** [comment]
@torkelo / @bergquist Please review. :)

**Comment by mattttt** [comment]
@utkarshcmu We can move that to the subnav once it's implemented (very soon). Will work with Torkelo to implement.

**Comment by utkarshcmu** [comment]
@mattttt any updates on this PR?

**Comment by mattttt** [comment]
HI @utkarshcmu, not yet, but shouldnt hold up this PR. It appearing in the main nav in master for the moment is kosher from my perspective.

**Comment by utkarshcmu** [comment]
@torkelo So, should we merge this PR as @mattttt is fine with the Snapshot menu appears in the main nav for now which is fine? :)

**Comment by torkelo** [comment]
I have not had time to review it yet after the changes were made

**Comment by utkarshcmu** [comment]
No, problem. Please review when you have time. :)

**Pull request merged**